content/realtime/metachannels.textile

```diff
@@ -115,17 +116,150 @@ metaChannel.subscribe('channel.region.inactive', (msg) => {
+- host := The host the request was made to.<br>__Type: @String@__
+- clientId := The ID of the client that made the request.<br>__Type: @String@__
+- requestId := The unique ID of the request.<br>__Type: @String@__
+- region := The region the request was made in.<br>__Type: @String@__
+- headers := The headers sent with the request.<br>__Type: @JSON Object@__
+- query := The details of the request query.<br>__Type: @JSON Object@__
+- channels := A list of channels the request was made against.<br>__Type: @Array@__
+- error := The details of any error received whilst making the request.<br>__Type: @JSON Object@__
+
+An example of a @request.refused@ event is:
+
+```[json]
+{
+  name: 'request.refused',
+  time: 1612286351217,
+  host: 'acme-realtime.ably.io',
+  clientId: '12345',
+  requestId: 'fbbcb0ab-fa56-47c4-bbd4-fccc22a271b8',
+  headers: {
+    ua: 'okhttp/3.12.3',
+  },
+  query: {
+    v='1.0',
+    format: 'msgpack',
+    lib: 'android-1.2.1'
+  },
+  channels: [
+    'myChannelA',
+  ],
+  error: {
+    message: 'unable to get channel',
+    statusCode: 404,
+    code: 40400,
+  },
+}
+```
+
+h2(#reactor). Reactor rules
 
-You can associate "Reactor rules":https://www.ably.com/integrations with metachannels in the same way as for regular channels; configure these using the "Message":/realtime/messages source and use a @channelFilter@ of, for example, @^\[meta\]log@. Normally the @channelFilter@ regex will only match normal, non-meta channels; however, a regex of the form @^\[qualifier\]pattern@ is special-cased to match channels with that qualifier. The @pattern@ is a regex as normal.
+You can associate "Reactor rules":https://www.ably.com/integrations with metachannels in the same way as for regular channels. Reactor rules are often used with metachannels so that events can be sent to "serverless endpoints":/general/events, "queues":/general/queues, and "streams":/general/firehose. To configure a reactor rule with a metachannel, use the "Message":/realtime/messages source and use a @channelFilter@ of, for example, @^\[meta\]log@. Normally the @channelFilter@ regex will only match normal, non-meta channels; however, a regex of the form @^\[qualifier\]pattern@ is special-cased to match channels with that qualifier. The @pattern@ is a regex as normal.
 
-If you want webhooks for channel lifecycle events, creating a "channel lifecycle webhook":/general/events#sources is an easy alternative to creating a message rule for the @[meta]channel.lifecycle@ metachannel; such webhooks will have simpler POST bodies as the lifecycle events won't be wrapped in "@Message@":/realtime/types#message objects.
+If you want webhooks for metachannel events, creating a "webhook":/general/events#sources is an easy alternative to creating a message rule for the metachannel; such webhooks will have simpler POST bodies as the lifecycle events won't be wrapped in "@Message@":/realtime/types#message objects.
```

Comment:
The sentence doesn't make sense now, but just dropping "If you want webhooks for metachannel events" wouldn't help much -- "Creating a webhook is an easy alternative to creating a message rule for the metachannel" still doesn't make much sense, a webhook is a kind of message rule.

The point of this paragraph is that a "channel lifecycle webhook" is a _special kind of webhook_ -- one with the rule source set to "channel.lifecycle". So if you want to get a webhook (or whatever) for every channel lifecycle event, creating a channel lifecycle webhook rule is an alternative to creating a message rule for every message on the channel lifecycle metachannel.

(And in future it will be a much better alternative, because that won't be subject to the rate-limiting you have by funneling everything through the metachannel)

Reply:
I've re-written this as a separate heading to make it clearer. Thanks for the explanation, Simon.